cdc/sinkv2/eventsink/txn/worker.go

```diff
@@ -67,49 +82,91 @@ func (w *worker) runBackgroundLoop() {
 		defer w.wg.Done()
 		defer func() {
 			if err := w.backend.Close(); err != nil {
-				log.Info("transaction sink backend close fail",
+				log.Info("Transaction sink backend close fail",
+					zap.String("changefeedID", w.changefeed),
+					zap.Int("workerID", w.ID),
 					zap.Error(err))
 			}
 		}()
-		w.timer = time.NewTimer(w.backend.MaxFlushInterval())
+		defer func() {
+			var r interface{}
+			if r = recover(); r == nil {
+				return
+			}
+			buf := make([]byte, 4096)
+			stackSize := runtime.Stack(buf, false)
+			buf = buf[:stackSize]
+			log.Error("Transaction sink worker panics", zap.Reflect("r", r), zap.Stack("stacktrace"))
+			w.errCh <- cerror.ErrMySQLWorkerPanic.GenWithStack("Transaction sink worker panics, stack: %v", string(buf))
+		}()
+		log.Info("Transaction sink worker starts",
+			zap.String("changefeedID", w.changefeed),
+			zap.Int("workerID", w.ID))
+
+		w.timer = time.NewTimer(w.flushInterval)
+		wantMoreCallbacks := make([]func(), 0, 1024)
+	LOOP:
 		for {
 			select {
+			case <-w.ctx.Done():
+				log.Info("Transaction sink worker exits as canceled",
+					zap.String("changefeedID", w.changefeed),
+					zap.Int("workerID", w.ID))
+				return
 			case <-w.stopped:
-				log.Info("Transaction sink backend worker exits expectedly",
+				log.Info("Transaction sink worker exits as closed",
+					zap.String("changefeedID", w.changefeed),
 					zap.Int("workerID", w.ID))
 				return
 			case txn := <-w.txnCh.Out():
 				metrics.ConflictDetectDuration.Observe(time.Since(txn.start).Seconds())
-				txn.wantMore()
-				if w.backend.OnTxnEvent(txn.txnEvent.TxnCallbackableEvent) && w.doFlush() {
-					log.Warn("Transaction sink backend exits unexceptedly")
-					return
+				wantMoreCallbacks = append(wantMoreCallbacks, txn.wantMore)
+				if w.backend.OnTxnEvent(txn.txnEvent.TxnCallbackableEvent) {
+					if w.doFlush() {
```

Comment:
Could you please add some comments to explain what happened and why we need to break or continue?
Also, should we add some logs here?

Reply:
If you mean the meanning of `doFlush` return value, It has already been commentted on `doFlush` itself.